tools/rosbag/src/player.cpp

```diff
@@ -56,21 +56,22 @@ using ros::Exception;
 
 namespace rosbag {
 
-ros::AdvertiseOptions createAdvertiseOptions(const ConnectionInfo* c, uint32_t queue_size) {
-    ros::AdvertiseOptions opts(c->topic, queue_size, c->md5sum, c->datatype, c->msg_def);
+ros::AdvertiseOptions createAdvertiseOptions(const ConnectionInfo* c, uint32_t queue_size, const std::string& prefix) {
+    ros::AdvertiseOptions opts(prefix+c->topic, queue_size, c->md5sum, c->datatype, c->msg_def);
```

Comment:
Please add spaces around the `+` operator to be consistent with the code style in this file.

Same below.